### DISCUSSION TITLE
How to run steps in one scenario step by step if one of them is async

### DISCUSSION THREAD

**Discussion by codehamal365** [getgauge/gauge#2173] [General]
for example.
## scenario example
* step one
* step two
* step three

step two is async step which will call api method and if step three need step two's result to assert. 
how to do that, can somebody give any suggestion.
btw the plugin is python.

**Comment by codehamal365** [comment]
@zabil I would appreciate it if you could give some suggestion

**Comment by zabil** [comment]
> step two is async step which will call api method and if step three need step two's result to assert.

To share values between steps you'll need to add values into a [data store](https://docs.gauge.org/writing-specifications.html?os=macos&language=python&ide=vscode#data-store).

However, this won't work predictably for async calls. You will have to call it synchronously or wait for it to complete.

**Reply by codehamal365** [reply]
OK.Thanks. Right now i use time sleep in python to solve this issue. Hopes gauge will support async way in future.
